anta/tests/connectivity.py

```diff
@@ -0,0 +1,48 @@
+"""
+Test functions related to various connectivity checks
+"""
+from __future__ import annotations
+
+import logging
+from typing import Any, Dict, cast
+
+from anta.models import AntaTest, AntaTestTemplate
+
+logger = logging.getLogger(__name__)
+
+
+class VerifyReachability(AntaTest):
+    """
+    Test network reachability to one or many destination IP(s).
+
+    Expected Results:
+        * success: The test will pass if all destination IP(s) are reachable.
+        * failure: The test will fail if one or many destination IP(s) are unreachable.
+        * error: The test will give an error if the destination IP(s) or the source interface/IP(s) are not provided as template_params.
+    """
+
+    name = "VerifyReachability"
+    description = "Test the network reachability to one or many destination IP(s)."
+    categories = ["connectivity"]
+    template = AntaTestTemplate(template="ping {dst} source {src} repeat 2")
+
+    @AntaTest.anta_test
+    def test(self) -> None:
+        """
+        Run VerifyReachability validation.
+        """
+
+        failures = []
+
+        for index, command in enumerate(self.instance_commands):
+            src, dst = (cast(Dict[str, str], command.template_params)["src"], cast(Dict[str, str], command.template_params)["dst"])
```

Comment:
so the answer is most certainly yes but do you need casting for the template_params as well here?

Reply:
Yes :)

`anta/tests/connectivity.py:38: error: Value of type "Optional[Dict[str, str]]" is not indexable  [index]`

Reply:
ok that's something it would be nice to get rid of :)